### PR TITLE
USWDS-Site: Added checkbox accessibility test page

### LINKED ISSUE #2799: CC a11y: publish checkbox checklist - create PR
## Summary
Publish the `usa-checkbox` accessibility test checklist items found in the [Critical Checklist spreadsheet](https://docs.google.com/spreadsheets/d/1fAJaObQHv7aIaeUpH4fTLmauhmugMtNCrbpZe4BJcPY/edit?usp=sharing) (Google Doc :lock:). 

The process for creating a new accessibility test page can be found in the [Creating accessibility checklists - process doc](https://docs.google.com/document/d/14WTCsdzgwpoz_Miztd66RmEGfwwSEUFCVN1DkSQITTE/edit#heading=h.fa3sv1ray42g) (Google Doc :lock:).

### PR DESCRIPTION
**Pull request by RachelCorsino**
# Summary

Added accessibility test page for checkbox component

## Related issue

Closes #2799 

## Preview link
Preview link:
[Checkbox Component Page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/rc-publish-checkbox-checklist/components/checkbox/#accessibility-test-status)
[Checkbox Accessibility Tests Page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/rc-publish-checkbox-checklist/components/checkbox/accessibility-tests/)

## Testing and review
Follow these steps:

1. Confirm that both the main component page and the accessibility tests page have the correct compliance tag at the top.
2. Confirm that the main component page links to the accessibility tests page in the side navigation.
3. Confirm that the main component page links to the accessibility tests page from the accessibility guidance section.
4. Confirm that the main component page shows the accessibility tests summary.
5. Check that accessibility tests page provides the correct counts for each test status type.
6. Confirm the test checklist summaries and data are accurate.
7. Confirm no visual issues.
8. Confirm there is an appropriate changelog entry on both the main component page and the accessibility tests page.

### CODE REVIEW

File: _components/checkbox/checkbox.md [hunk 1]
```diff
@@ -10,16 +10,8 @@ permalink: /components/checkbox/
...
-  href: '#checkbox-preview'
-- text: Code
-  href: '#checkbox-code'
-- text: Guidance
-  href: '#checkbox-guidance'
-- text: Package
-  href: '#checkbox-package'
-- text: Latest updates
-  href: '#changelog'
+- text: checkbox accessibility tests
```
Comment (amyleadem) [inline _components/checkbox/checkbox.md:13]: @RachelCorsino 

**issue (blocking):** Little typo
```suggestion
- text: Checkbox accessibility tests
```

File: _data/accessibility-tests/checkbox.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    test_status: pass
+    test_type: keyboard
+    version_tested: 3.8.1
+    wcag_criterion: 2.1.1
+# Screen reader tests
+  - summary: Screen reader announces question before any checkbox and then elements of the checkbox.
+    summary_additional: When listening to the checkbox button using a screen reader, you hear the question related to the checkbox read first. Then you will hear the checkbox group and labels within the checkbox and their checked state. 
+    test_status: pass
+    test_type: screen_reader
+    version_tested: 3.8.1
```
Comment (amyleadem) [inline _data/accessibility-tests/checkbox.yml:41]: @RachelCorsino 

**issue (blocking):** Looks like this was last tested on 04/05/2024. At that time, [only 3.8.0 had been released](https://github.com/uswds/uswds/releases). It looks like this is true for all the criteria. Can you update all the 3.8.1 references to 3.8.0 in this file?
```suggestion
    version_tested: 3.8.0
```

File: _data/accessibility-tests/checkbox.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    wcag_criterion: 2.4.6
+# Keyboard navigation tests
+  - summary: Focus indicator on checkbox is clearly visible.
+    summary_additional: When you use a keyboard to focus on a checkbox group, you will see a visible outline or other clear indication of focus. 
+    test_status: pass
+    test_type: keyboard
+    version_tested: 3.8.1
+    wcag_criterion: 3.2.1
+  - summary: Checkbox focus follows logical content order on desktop.
```
Comment (amyleadem) [inline _data/accessibility-tests/checkbox.yml:24]: @finekatie @amycole501 @alex-hull 

Can we remove "on desktop" from this summary? It seems like it would be applicable at all screen widths.

```suggestion
  - summary: Checkbox focus follows logical content order.
```

File: _data/accessibility-tests/checkbox.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    summary_additional: When you tab through the checkbox with a keyboard, the focus stays in order of the content as intended. 
+    test_status: pass
+    test_type: keyboard
+    version_tested: 3.8.1
+    wcag_criterion: 2.4.3
+  - summary: Checkbox focus can be moved and options can be selected.
```
Comment (amyleadem) [inline _data/accessibility-tests/checkbox.yml:30]: @finekatie 
cc: @alex-hull @amycole501 

**suggestion (non-blocking):** Wondering if we can simplify this one to focus more on the keyboard interaction. Maybe something like this?
```suggestion
  - summary: Users can access and select checkbox items using only a keyboard.
```

File: _data/accessibility-tests/checkbox.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    version_tested: 3.8.1
+    wcag_criterion: 3.2.1
+  - summary: Checkbox focus follows logical content order on desktop.
+    summary_additional: When you tab through the checkbox with a keyboard, the focus stays in order of the content as intended. 
+    test_status: pass
+    test_type: keyboard
+    version_tested: 3.8.1
+    wcag_criterion: 2.4.3
+  - summary: Checkbox focus can be moved and options can be selected. 
+    summary_additional: When you use a keyboard to navigate to the checkbox, you are able to move focus and select options, using different keys.
```
Comment (amyleadem) [inline _data/accessibility-tests/checkbox.yml:31]: @finekatie
cc: @alex-hull @amycole501

**suggestion (non-blocking):** I found myself asking "which different keys"? Wondering if we can remove "using different keys" from the summary OR specifying which keys. I tried my hand at a couple possibilities below:

Option 1:
```suggestion
    summary_additional: When you use a keyboard to navigate to the checkbox, you are able to move focus and select options.
```

Option 2:
```suggestion
    summary_additional: When you use a keyboard to navigate to the checkbox, you are able to move focus using `tab` and can select options with `spacebar`.
```

File: _data/accessibility-tests/checkbox.yml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    version_tested: 3.8.1
+    wcag_criterion: 2.4.3
+  - summary: Checkbox focus can be moved and options can be selected. 
+    summary_additional: When you use a keyboard to navigate to the checkbox, you are able to move focus and select options, using different keys. 
+    test_status: pass
+    test_type: keyboard
+    version_tested: 3.8.1
+    wcag_criterion: 2.1.1
+# Screen reader tests
+  - summary: Screen reader announces question before any checkbox and then elements of the checkbox.
```
Comment (amyleadem) [inline _data/accessibility-tests/checkbox.yml:37]: @finekatie
cc: @alex-hull @amycole501

**suggestion (non-blocking):** When reading this, I got stuck trying to figure out what the "question" here was. Can we rephrase it to be something like this? 

```suggestion
  - summary: Screen reader announces the checkbox group label and then announces each checkbox.
```

**Review by amyleadem — CHANGES_REQUESTED** [review]
@RachelCorsino Thanks for getting this set up so quickly! I tagged you in a couple of comments below - they should be easy fixes. I also tagged @finekatie @alex-hull and @amycole501 in some comments about some of the summaries. Let me know if you have any questions!

File: _data/accessibility-tests/checkbox.yml (hunk 3 above)
Comment (alex-hull) [reply _data/accessibility-tests/checkbox.yml:24]: Yes, I think it is fine to remove that.

File: _data/accessibility-tests/checkbox.yml (hunk 4 above)
Comment (alex-hull) [reply _data/accessibility-tests/checkbox.yml:30]: Yes! I think we can make this change!

File: _data/accessibility-tests/checkbox.yml (hunk 5 above)
Comment (alex-hull) [reply _data/accessibility-tests/checkbox.yml:31]: I think the second option above works with what we were going with. I would want people to know which keys should be working for this specific interaction.

File: _data/accessibility-tests/checkbox.yml (hunk 6 above)
Comment (alex-hull) [reply _data/accessibility-tests/checkbox.yml:37]: Ooh, valid. It may also not always be a question the user is selecting. I think the suggested change works.

File: _components/checkbox/checkbox.md (hunk 1 above)
Comment (RachelCorsino) [reply _components/checkbox/checkbox.md]: Resolved in [5808ba8](https://github.com/uswds/uswds-site/pull/2803/commits/5808ba81a1e73b90ad5129c8a440f7a27c6ad4a9)

File: _data/accessibility-tests/checkbox.yml (hunk 2 above)
Comment (RachelCorsino) [reply _data/accessibility-tests/checkbox.yml]: Resolved in [5808ba8](https://github.com/uswds/uswds-site/pull/2803/commits/5808ba81a1e73b90ad5129c8a440f7a27c6ad4a9)

File: _data/accessibility-tests/checkbox.yml (hunk 3 above)
Comment (RachelCorsino) [reply _data/accessibility-tests/checkbox.yml]: Resolved in [5808ba8](https://github.com/uswds/uswds-site/pull/2803/commits/5808ba81a1e73b90ad5129c8a440f7a27c6ad4a9)

File: _data/accessibility-tests/checkbox.yml (hunk 4 above)
Comment (RachelCorsino) [reply _data/accessibility-tests/checkbox.yml]: Resolved in [5808ba8](https://github.com/uswds/uswds-site/pull/2803/commits/5808ba81a1e73b90ad5129c8a440f7a27c6ad4a9)

File: _data/accessibility-tests/checkbox.yml (hunk 5 above)
Comment (RachelCorsino) [reply _data/accessibility-tests/checkbox.yml]: Resolved in [5808ba8](https://github.com/uswds/uswds-site/pull/2803/commits/5808ba81a1e73b90ad5129c8a440f7a27c6ad4a9)

File: _data/accessibility-tests/checkbox.yml (hunk 6 above)
Comment (RachelCorsino) [reply _data/accessibility-tests/checkbox.yml]: Resolved in [5808ba8](https://github.com/uswds/uswds-site/pull/2803/commits/5808ba81a1e73b90ad5129c8a440f7a27c6ad4a9)

**Comment by finekatie** [comment]
Hi @amyleadem , I saw that I was added to this but I appear to now be removed. Do you still need my review? Apologies for not being able to get to it sooner.

File: _data/accessibility-tests/checkbox.yml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+test_items:
+# General tests
+  - summary: Visible instructions explain checkbox requirements. 
+    summary_additional: The checkbox has a visible label above it to indicate its purpose. For example, "Select your state."
+    test_status: pass
+    test_type: general
+    version_tested: 3.8.0
+    wcag_criterion: 3.3.2
+  - summary: Radio button labels are descriptive, brief and clear.
+    summary_additional: When you view radio buttons in a checkbox, all labels have short descriptions that are easy to understand and act on.
```
Comment (amyleadem) [inline _data/accessibility-tests/checkbox.yml:12]: @finekatie 
cc @alex-hull @amycole501 

Looks like radio button references were accidentally included here. Recommend we borrow language from the [Select component page](https://designsystem.digital.gov/components/select/accessibility-tests/): 
```suggestion
  - summary: Checkbox labels are descriptive, brief and clear.
    summary_additional: When you view or interact with the checkbox component on your page, all labels have short descriptions that are easy to understand and act on. 
```

Comment (amyleadem) [inline _data/accessibility-tests/checkbox.yml:12]: @finekatie 
cc @alex-hull @amycole501 

Looks like radio button references were accidentally included here. Recommend we borrow language from the [Select component page](https://designsystem.digital.gov/components/select/accessibility-tests/): 
```suggestion
  - summary: Checkbox labels are descriptive, brief and clear.
    summary_additional: When you view or interact with the checkbox component on your page, all labels have short descriptions that are easy to understand and act on. 
```

**Review by amyleadem — CHANGES_REQUESTED** [review]
Looks good @RachelCorsino! I discovered one other item to fix, highlighted in the comment below. 

@finekatie, I just tagged you as a reviewer on this PR. We made some updates to the summaries. Can you confirm that they all meet standards?

File: _data/accessibility-tests/checkbox.yml (hunk 7 above)
Comment (finekatie) [reply _data/accessibility-tests/checkbox.yml:12]: I wondered about that when I saw it, but I figured there must be a rhyme/reason. Shoulda trusted my instincts! This looks good to me.

**Review by finekatie — APPROVED** [review]
I approve!

**Review by amyleadem — APPROVED** [review]
Looks good to me!

**Comment by amycole501** [comment]
The first focus indicator check under "keyboard" has the wrong WCAG criteria. It should be WCAG Focus visible AA https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html

**Comment by RachelCorsino** [comment]
> The first focus indicator check under "keyboard" has the wrong WCAG criteria. It should be WCAG Focus visible AA https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html

Resolved in [6209524](https://github.com/uswds/uswds-site/pull/2803/commits/62095249f73c46d4e6d3312103d35f4a84871fb4).

**Review by amycole501 — APPROVED** [review]

File: _data/changelogs/component-checkbox-accessibility.yml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+title: Checkbox
+type: component
+items:
+ - date: NNNN-NN-NN
```
Comment (amyleadem) [inline _data/changelogs/component-checkbox-accessibility.yml:4]: > [!Important]
> We need to update the changelog dates before merge

Comment (thisisdano) [reply _data/changelogs/component-checkbox-accessibility.yml:4]:
```suggestion
 - date: 2024-09-18
```

File: _data/changelogs/component-checkbox.yml [hunk 9]
```diff
@@ -2,6 +2,11 @@ title: Checkbox
 type: component
 changelogURL:
 items:
+  - date: NNNN-NN-NN
```
Comment (thisisdano) [inline _data/changelogs/component-checkbox.yml:5]:
```suggestion
  - date: 2024-09-18
```

**Review by thisisdano — APPROVED** [review]

**Review by thisisdano — APPROVED** [review]

**Comment by thisisdano** [comment]
Merging over htmlproofer false positives in Circle.
